src/main/java/cloud/orbit/actors/cluster/impl/RedisConnectionManager.java

```diff
@@ -140,22 +140,34 @@ public void subscribeToChannel(final String channelId, final MessageListener<Obj
     public void sendMessageToChannel(final String channelId, final Object msg)
     {
             final List<RedisOrbitClient> localMessagingClients = messagingClients.stream().filter((e) -> e.isConnectied()).collect(Collectors.toList());
-            final int activeClientCount = localMessagingClients.size();
-            if(activeClientCount > 0)
-            {
-                final int randomId = ThreadLocalRandom.current().nextInt(activeClientCount);
-                final RedissonClient client = localMessagingClients.get(randomId).getRedissonClient();
-
-                client.getTopic(channelId).publishAsync(msg).exceptionally((e) ->
-                {
-                    logger.error("Error sending message", e);
-                    return 0L;
-                });
-            }
-            else
-            {
-                throw new UncheckedException("No Redis messaging instances available.");
-            }
+            sendMessageToChannel(channelId, msg, localMessagingClients, 0);
+    }
+
+    private void sendMessageToChannel(final String channelId, final Object msg, final List<RedisOrbitClient> localMessagingClients, final int attempt) {
+        final int activeClientCount = localMessagingClients.size();
+        if(activeClientCount > 0)
+        {
+            final int randomId = ThreadLocalRandom.current().nextInt(activeClientCount);
+            final RedissonClient client = localMessagingClients.get(randomId).getRedissonClient();
+
+            client.getTopic(channelId).publishAsync(msg)
+                    .whenComplete((x, e) -> {
+                       if(e != null) {
+                           logger.error("Failed to send message to channel '{}'", channelId, e);
+                       } else if(x == 0) {
+                           if(attempt >= redisClusterConfig.getMessageResendAttempts()) {
+                               logger.error("Failed to send message to channel '{}' after {} attempts.", channelId, attempt);
+                           } else {
+                               logger.warn("Failed to send message to channel '{}' on attempt {}. Retrying...", channelId, attempt);
+                               sendMessageToChannel(channelId, msg, localMessagingClients, attempt + 1);
+                           }
+                       }
+                    });
+        }
+        else
+        {
+            logger.error("Failed to send message to channel '{}', no redis messaging instances were available after {} attempts.", channelId, attempt);
```

Comment:
Seems that this should just be a generic error message about no active clients available. I don't think this matters on the number of attempts or that it was trying to send to a specific channel. More that the node doesn't know about any messaging nodes to send too.

Reply:
I do think this should include that info because I refactored so the list actually gets smaller, so it's relevant if it failed on attempt 1 (i.e no nodes available) vs attempt 7, because that suggests the remote node is not subscribed anywhere.

Reply:
I don't see that change where the list is getting smaller? Would we really be wanting to make the list smaller? The messaging node might be fine just that the backend host didn't register itself on that node so there is no consumer. Would seem odd to remove the client. Unless it is just a local collection for the attempts then that seems more reasonable. I was wondering that when reviewing how on the recursion we don't end up picking the same node as we did the previous time.

Reply:
```
final RedissonClient client = localMessagingClients.remove(randomId).getRedissonClient();
```

Yes, it's just a local list for that try. It has always been a local list of "active" connections at the the time the message was sent so this is just manipulating that list.

Reply:
Ah... missed that. Thought it was doing a `get`. 👍